Wrappers/Python/ccpi/viewer/utils/io.py

```diff
@@ -42,3 +66,765 @@ def SaveRenderToPNG(render_window, filename):
     writer.SetFileName(saveFilename)
     writer.SetInputConnection(w2if.GetOutputPort())
     writer.Write()
+
+
+# TODO:
+# supporting tiffs
+# write out other filetypes
+
+
+class ImageReader(object):
+    '''
+    Generic reader for reading to vtkImageData
+    Currently reads: HDF5, MetaImage, Numpy, Raw
+    Later will support TIFFs
+    Supports resampling OR cropping the dataset whilst
+    reading.
+    Currently doesn't support both cropping and resampling
+    If set both to true then it resamples and doesn't crop
+    '''
+
+    def __init__(self,
+                 file_name=None,
+                 resample=True,
+                 target_size=512**3,
+                 crop=False,
+                 target_z_extent=None,
+                 resample_z=False,
+                 raw_image_attrs=None,
+                 hdf5_dataset_name="entry1/tomo_entry/data/data",
+                 log_file=None):
+        '''
+        Constructor
+
+        Parameters
+        ----------
+        file_name: os.path or string, default None
+            file name to read
```

Comment:
In case of TIFF files, either the directory to the TIFF files, or the path to one such file is needed.